packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -63,53 +63,53 @@ export const typeOrValueSpecifierSchema: JSONSchema4 = {
         },
       },
       required: ['from', 'name'],
+      type: 'object',
     },
     {
-      type: 'object',
       additionalProperties: false,
       properties: {
         from: {
-          type: 'string',
           enum: ['lib'],
+          type: 'string',
         },
         name: {
           oneOf: [
             {
               type: 'string',
             },
             {
-              type: 'array',
-              minItems: 1,
-              uniqueItems: true,
               items: {
                 type: 'string',
               },
+              minItems: 1,
+              type: 'array',
+              uniqueItems: true,
             },
           ],
         },
       },
       required: ['from', 'name'],
+      type: 'object',
     },
     {
-      type: 'object',
       additionalProperties: false,
       properties: {
         from: {
-          type: 'string',
           enum: ['package'],
+          type: 'string',
```

Comment:
this is the same "type prop should be first" problem as ast-spec.
minor nit though.

Reply:
👍 resolved in other PRs, will pull in fixes here